src/omnisharp/experimentalOmnisharp.Downloader.ts

```diff
@@ -0,0 +1,72 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { PackageManager, Package } from '../packages';
+import { PlatformInformation } from '../platform';
+import { Logger } from '../logger';
+import TelemetryReporter from 'vscode-extension-telemetry';
+import { GetPackagesFromVersion } from './experimentalOmnisharp.PackageCreator';
+import { GetDependenciesAndDownloadPackages, GetStatus, GetAndLogPlatformInformation, ReportInstallationError, SendInstallationTelemetry } from '../experimentalOmnisharp.Download.Helper';
+
+export class ExperimentalOmnisharpDownloader {
+
+    public constructor(
+        private channel: vscode.OutputChannel,
+        private logger: Logger,
+        private reporter: TelemetryReporter /* optional */,
+        private packageJSON: any) {
+    }
+
+    public async DownloadAndInstallExperimentalVersion(version: string, serverUrl: string, installPath: string) {
```

Comment:
What is the plan to unify this with the code in https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/CSharpExtDownloader.ts where this is copied and pasted from?

Reply:
Once this feature has been added and tested satisfactorily, we plan to unify the two in a separate PR